benchmarks/registry.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const _ = require('lodash');
+
+module.exports = setupRegistrySuite;
+
+function setupRegistrySuite(client, { add }) {
+	const registry = new client.Registry();
+
+	const histogram = new client.Histogram({
+		name: 'histogram',
+		help: 'histogram',
+		labelNames: ['a', 'b', 'c', 'd'],
+		registers: [registry]
+	});
+
+	const labelValueCount = 2;
+	_.times(labelValueCount, a => {
+		_.times(labelValueCount, b => {
+			_.times(labelValueCount, c => {
+				_.times(labelValueCount, d => {
+					const end = histogram.startTimer();
+
+					end({ a, b, c, d });
+				});
+			});
+		});
+	});
+
+	add('getMetricsAsJSON', () => {
+		registry.getMetricsAsJSON();
+	});
+
+	add('metrics', () => {
+		registry.metrics();
+	});
+}
```

Comment:
Is it worth adding something like:

        add('histogramObserve', () => {
                histogram.observe(1);
        });

(I've tinkered with performance changes that hit the observe() call in exchange for scrape benefits.  This would trigger a fail, but given the consequences of that I think a fail on observe() performance is good as it makes for a debate on whether or not that is acceptable.)

Reply:
I think adding a suite for updating a metric, and not just serializing them, is very much valuable!

Reply:
Agreed! I'll add that to this PR. Currently I am working on adding logic to allow acceptable margin of error to speed differences (to account fro no-op performance PRs and run variance)